### ISSUE TITLE
Accuracy null and low loss

### ISSUE THREAD

**Issue by Erickira3** [tflearn/tflearn#833]
Hello everyone and thank you for looking at my question. 

I have data made of date (one entry per minute) for the first column and congestion (value, between 0 and 200) for the 2nd. My goal is to feed it to my neural network and so be able to predict for the next week the congestion at each minute (my dataset is more than 10M of entry, I shouldn't have problem of lack of data for training).

Problem: I have an error (may be linked to me using the wrong loss/optimizer). When using my code (10 000 data of training for now) I have an accuracy of 0, a low loss (0.00X) and a bad prediction (not even close to the reality). Do you have any idea of where it could come from (I use Adam as an optimizer, Mean_square for loss and linear for activation)?

Here is an example of what I got 

> Training Step: 7100 | total loss: 0.00304 | time: 1.385s | Adam | epoch: 100 | loss: 0.00304 - binary_acc: 0.0000 | val_loss: 0.00260 - val_acc: 0.0000 -- iter: 8999/8999

My code look like that:
```
from __future__ import print_function
import numpy
import numpy as np
import tflearn
from pandas import DataFrame
from pandas import Series
from pandas import concat
from pandas import read_csv
from sklearn.preprocessing import MinMaxScaler
from sklearn.metrics import mean_squared_error
from math import sqrt

import datetime

# Preprocessing function
from tflearn import Accuracy


def preprocess(data):
    return np.array(data, dtype=np.int32)

def parser(x):
    return datetime.datetime.strptime(x, '%Y-%m-%d %H:%M:%S')


# frame a sequence as a supervised learning problem
def timeseries_to_supervised(data, lag=1):
    df = DataFrame(data)
    columns = [df.shift(i) for i in range(1, lag + 1)]
    columns.append(df)
    df = concat(columns, axis=1)
    df.fillna(0, inplace=True)
    return df


def difference(dataset, interval=1):
    diff = list()
    for i in range(interval, len(dataset)):
        value = dataset[i] - dataset[i - interval]
        diff.append(value)
    return Series(diff)


# invert differenced value
def inverse_difference(history, yhat, interval=1):
    return yhat + history[-interval]


# scale train and test data to [-1, 1]
def scale(train, test):
    # fit scaler
    scaler = MinMaxScaler(feature_range=(-1, 1))
    scaler = scaler.fit(train)
    # transform train
    train = train.reshape(train.shape[0], train.shape[1])
    train_scaled = scaler.transform(train)
    # transform test
    test = test.reshape(test.shape[0], test.shape[1])
    test_scaled = scaler.transform(test)
    return scaler, train_scaled, test_scaled


# inverse scaling for a forecasted value
def invert_scale(scaler, X, value):
    new_row = [x for x in X] + [value]
    array = numpy.array(new_row)
    array = array.reshape(1, len(array))
    inverted = scaler.inverse_transform(array)
    return inverted[0, -1]


def fit_lstm(train, batch_size, nb_epoch, neurons):
    X, y = train[0:-1], train[:, -1]
    X = X[:, 0].reshape(len(X), 1)
    y = y.reshape(len(y), 1)
    # Build neural network
    net = tflearn.input_data(shape=[None, 1])
    net = tflearn.fully_connected(net, 1, activation='linear')
    acc = Accuracy()
    net = tflearn.regression(net, optimizer='adam', learning_rate=0.001,
                             loss='mean_square', metric=acc)
    # Define model
    model = tflearn.DNN(net)
    print (X.shape)
    model.fit(X, y, n_epoch=nb_epoch, batch_size=batch_size, shuffle=False, show_metric=True)

    return model


# make a one-step forecast
def forecast_lstm(model, X):
    X = X.reshape(len(X), 1)
    yhat = model.predict(X)
    return yhat[0, 0]

# Load CSV file, indicate that the first column represents labels
data = read_csv('nowcastScaled.csv', header=0, parse_dates=[0], index_col=0, squeeze=True, date_parser=parser)


# transform data to be stationary
raw_values = data.values
diff_values = difference(raw_values, 1)

# transform data to be supervised learning
supervised = timeseries_to_supervised(diff_values, 1)
supervised_values = supervised.values

# split data into train and test-sets
train, test = supervised_values[0:10000], supervised_values[10000:11000]

# transform the scale of the data
scaler, train_scaled, test_scaled = scale(train, test)

# fit the model
lstm_model = fit_lstm(train_scaled, 16, 15, 1)
lstm_model.save('ModelTrainTfLearn')

# forecast the entire training dataset to build up state for forecasting
train_reshaped = train_scaled[:, 0].reshape(len(train_scaled), 1)
predictiont = lstm_model.predict(train_reshaped)
print("Prediction: %s" % str(predictiont[0][:3]))  # only show first 3 probas

# walk-forward validation on the test data
predictions = list()
error_scores = list()
for i in range(len(test_scaled)):
    # make one-step forecast
    X, y = test_scaled[i, 0:-1], test_scaled[i, -1]
    yhat = forecast_lstm(lstm_model, X)
    # X, y = test_scaled[i, 0:-1], test_scaled[i, -1]
    # invert scaling
    yhat = invert_scale(scaler, X, yhat)
    # # invert differencing
    yhat = inverse_difference(raw_values, yhat, len(test_scaled) + 1 - i)
    # store forecast
    predictions.append(yhat)
rmse = sqrt(mean_squared_error(raw_values[10000:11000], predictions))
print('%d) Test RMSE: %.3f' % (1, rmse))
error_scores.append(rmse)
print (predictions)
print (raw_values[10000:11000])
```
What I'm trying to do is a time series prediction, is tflearn good for that or is it just my way to do it that is wrong?

Thank your for your attention and help.

(Linked is the file I'm using)
[nowcastScaled.zip](https://github.com/tflearn/tflearn/files/1141334/nowcastScaled.zip)

**Comment by Erickira3** [comment]
It seem like it is because my network architecture isn't converging. Would you have an idea of what could cause that?
